### ISSUE TITLE
Message encryption for Memberlist hardcoded to AES-128

### ISSUE THREAD

**Issue by rranjan3** [metallb/metallb#1982]
[Hashicorp Memberlist](https://github.com/hashicorp/memberlist) supports AES-128, AES-192 and AES-256 for message encryption. This can be found here - https://github.com/hashicorp/memberlist/blob/v0.5.0/config.go#L187.

A review of codebase suggests, hardcoding this to AES-128 only. This is essentially controlled by the size of the secret being used. For the latest code it can be observed that [the size is 16](https://github.com/metallb/metallb/blob/main/internal/speakerlist/speakerlist.go#L79).
```
		sha := sha256.New()
		mconfig.SecretKey = sha.Sum([]byte(secret))[:16]
```

Is this choice on purpose ? Can this be made configurable so that the choice of symmetric encryption algorithm can be made ? AES-192 & AES-256 might be a better option if given a choice.

**Comment by fedepaol** [comment]
Yes, this sounds doable or we can even default to the larger size.

**Comment by AzraelSec** [comment]
Hi @fedepaol 👋🏻 If it works for you, I'd like to pick this up.

> Yes, this sounds doable or we can even default to the larger size.

Do you prefer me to just hard-code the key length to the longer available value, or would you also like a customizable value? Thanks!

**Comment by fedepaol** [comment]
I don't see much value in making it configurable, just go for the longer

**Comment by fedepaol** [comment]
@AzraelSec are you still interested in working on this?

**Comment by AzraelSec** [comment]
Yes, @fedepaol, I am. Sorry for the long delay - I picked this up at the wrong moment. Is it acceptable for you to assign this issue to me for two days more? I should be able to work on this today/tomorrow. If I can't, I'll ping you asap so you can unassign it.

**Comment by fedepaol** [comment]
No rush! I was just checking :-)
